### PR TITLE
PR for #249

### PR DESCRIPTION
**Pull request by hirako2000**
- [x] Adds AnimationModule 
- [x] Adds example with space alien animation frame 
- [x] Adds some rock textures

### CODE REVIEW

File: examples/animation/alien/script.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  useCustomMaterial: true,
+  material: new THREE.MeshStandardMaterial({
+    shading: THREE.SmoothShading,
+    skinning: true,
+    emissiveIntensity: 1,
+    metalness: 1
+  }),
+
+  modules: [textureModule, animationModule]
+});
```
Comment (sasha240100) [inline examples/animation/alien/script.js]: newline

File: examples/animation/alien/script.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    distance: 20
+  },
+
+  shadow: {
+    far: 30,
+    near: 2
+  },
+
+  position: [-1, 8, 5]
+});
```
Comment (sasha240100) [inline examples/animation/alien/script.js]: \n

File: examples/animation/alien/script.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+whitePointLight.addTo(world);
+
+const floorTextureRepeat = new THREE.Vector2(15, 15);
```
Comment (sasha240100) [inline examples/animation/alien/script.js]: \n

File: examples/animation/alien/script.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+new WHS.Box({
+  geometry: {
+    width: 80,
+    height: 80,
+    depth: 0.1
+  },
+  position: [0, 0, 0],
```
Comment (sasha240100) [inline examples/animation/alien/script.js]: 81, 82: \n

File: examples/animation/alien/script.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+      url: `${process.assetsPath}/textures/rock/gbump.jpg`,
+      repeat: floorTextureRepeat,
+      type: 'bumpMap'
+    }, {
+      url: `${process.assetsPath}/textures/rock/groughness.jpg`,
+      repeat: floorTextureRepeat,
+      type: 'roughnessMap'
+    })
+  ]
+
```
Comment (sasha240100) [inline examples/animation/alien/script.js]: rm line

File: examples/animation/alien/script.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    x: 10,
+    y: 30,
+    z: 10
+  }
+}).addTo(world);
+
+UTILS.addAmbient(world, 0.1);
+
+new WHS.Loop(() => {
+  animationModule.update();
```
Comment (sasha240100) [inline examples/animation/alien/script.js]: Why not using `Loop` and `update()` inside module? 

`const animation = new WHS.AnimationModule(app, [isDeferred])` <- pass app inside constructor.

If `isDeferred` is **true** - then we don't start the loop by default. And user can do that manually with `animation.loop.start();`

File: src/modules/mesh/AnimationModule.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+export class AnimationModule {
+  constructor(params = {}) {
+    this.params = Object.assign(params);
+    this.clock = new Clock();
+  }
+
+  play(clipName) {
+    const clip = AnimationClip.findByName(this.clips, clipName);
+    const action = this.mixer.clipAction(clip);
```
Comment (sasha240100) [inline src/modules/mesh/AnimationModule.js]: 15-16: \n

File: src/modules/mesh/AnimationModule.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+  update() {
+    if (this.mixer) this.mixer.update(this.clock.getDelta());
+  }
+
+  bridge = {
+    mesh(mesh, self) {
+      mesh.geometry.skeleton = mesh.skeleton;
+      self.mixer = new AnimationMixer(mesh.geometry);
+      self.clips = mesh.geometry.animations;
```
Comment (sasha240100) [inline src/modules/mesh/AnimationModule.js]: \n

File: src/modules/mesh/AnimationModule.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    action.play();
+  }
+
+  update() {
+    if (this.mixer) this.mixer.update(this.clock.getDelta());
+  }
+
+  bridge = {
+    mesh(mesh, self) {
+      mesh.geometry.skeleton = mesh.skeleton;
```
Comment (sasha240100) [inline src/modules/mesh/AnimationModule.js]: \n

File: src/modules/mesh/__tests__/AnimationModule.spec.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
+import '../../../polyfill';
```
Comment (sasha240100) [inline src/modules/mesh/__tests__/AnimationModule.spec.js]: \n

File: src/modules/mesh/__tests__/AnimationModule.spec.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+import {Clock, AnimationMixer} from 'three';
+import {Box} from '../../../index';
+import {AnimationModule} from '../AnimationModule';
+
+const box = new Box();
+box.skeleton = ['skel'];
```
Comment (sasha240100) [inline src/modules/mesh/__tests__/AnimationModule.spec.js]: \n

File: src/modules/mesh/AnimationModule.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  play(clipName) {
+    const clip = AnimationClip.findByName(this.clips, clipName);
+    const action = this.mixer.clipAction(clip);
+    action.play();
+  }
+
+  update() {
+    if (this.mixer) this.mixer.update(this.clock.getDelta());
+  }
+
```
Comment (sasha240100) [inline src/modules/mesh/AnimationModule.js]: Let's add: 

```js
manager(manager) {
  manager.define('animation');
}
```

That way it's API can be used by other modules with `manager.use('animation')  <- returns module scope`

File: examples/animation/alien/script.js [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+  }),
+  new WHS.OrbitControlsModule(),
+  new WHS.ResizeModule()
+]);
+
+const animationModule = new WHS.AnimationModule();
+
+const characterTexturePath = `${process.assetsPath}/textures/space-alien/`;
+const textureModule = new WHS.TextureModule({
+  url: characterTexturePath + `diffuse.png`,
```
Comment (sasha240100) [inline examples/animation/alien/script.js]:
```js
`${characterTexturePath}diffuse.png`
``` 
?

File: examples/animation/alien/script.js [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+      physicallyCorrectLights: true
+    }
+  }),
+  new WHS.OrbitControlsModule(),
+  new WHS.ResizeModule()
+]);
+
+const animationModule = new WHS.AnimationModule();
+
+const characterTexturePath = `${process.assetsPath}/textures/space-alien/`;
```
Comment (sasha240100) [inline examples/animation/alien/script.js]: switch 20<->21

File: src/modules/mesh/AnimationModule.js (hunk 12 above)
Comment (hirako2000) [inline src/modules/mesh/AnimationModule.js]: there could be multiple animation module

Comment (sasha240100) [inline src/modules/mesh/AnimationModule.js]: You need to make it to be only one. Like it should handle multiple animations

**Pull request merged**